### PR TITLE
feat(aapd-861): update appeal and user link if role changes

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-861

## Description of change

Updates the role a user has on an appeal when isOriginalAppellant changes
Integration tests take into account a shared DB and parallel running

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeals-service-api/src/services/appeal.service.js [hunk 1]
```diff
@@ -141,6 +150,74 @@ async function updateAppeal(id, appealUpdate) {
 	return updatedAppeal;
 }
 
+/**
+ * @param {*} appealUpdate - appeal to update
+ * @param {*} appeal - existing appeal
+ * @returns {Promise<void>}
+ */
+async function linkToUser(appealUpdate, appeal) {
+	const updateLinkForS76 =
```
Comment (beejjacobs) [inline packages/appeals-service-api/src/services/appeal.service.js:159]: Meant to be S78?

File: packages/appeals-service-api/src/services/appeal.service.js [hunk 2]
```diff
@@ -141,6 +150,74 @@ async function updateAppeal(id, appealUpdate) {
...
+		appealUpdate?.contactDetailsSection?.isOriginalApplicant !== undefined &&
+		(appeal?.contactDetailsSection?.isOriginalApplicant === undefined ||
+			appealUpdate?.contactDetailsSection?.isOriginalApplicant !==
+				appeal?.contactDetailsSection?.isOriginalApplicant);
```
Comment (beejjacobs) [inline packages/appeals-service-api/src/services/appeal.service.js:163]: I find this hard to follow with the nesting. What about this? (similar for HAS)

even this is a bit long - they're awkward long names!

```
const contactDetailsIsOriginalApplicant = appeal?.contactDetailsSection?.isOriginalApplicant;
const updateContactDetailsIsOriginalApplicant = appealUpdate?.contactDetailsSection?.isOriginalApplicant;
const updateLinkForS76 = updateContactDetailsIsOriginalApplicant  !== undefined &&
		(updateContactDetailsIsOriginalApplicant  !== contactDetailsIsOriginalApplicant );
```

**Review by beejjacobs — APPROVED** [review]
Might be a typo in there?

Generally looks good!

File: packages/appeals-service-api/src/services/appeal.service.js (hunk 1 above)
Comment (bryn-pins) [reply packages/appeals-service-api/src/services/appeal.service.js:159]: ugh I get this wrong all the time, used to work on 96 and now confuse the numbers

**Review by beejjacobs — APPROVED** [review]
LGTM!
